### PR TITLE
Added functionality to return promotion via status

### LINKED ISSUE #41: Add capability to query promotions by it's Status in the List Method
**As a** User
**I need** to fetch active promotions through the existing List endpoint
**So that** I can use the active promotions

**Details & Assumptions:** 
* The base URL would be `/promotions?status=True`
* The HTTP method should be GET
* The URL should return a JSON response of active promotions

**Acceptance Criteria:**

```gherkin
Given the service is available
When an error occurs while hitting the endpoint
Then a proper HTTP Status Code depicting the type of error should be returned
When I call the URL`/promotions?status=True` with the HTTP GET method
Then It should return details of all the active promotions in the format: 
```

```
{
promotions: {
  [{
    id: <promotion id>,
    name: <promotion name>,
    type: <promotion type>,
    description: <promotion description>,
    promotion_value: <promotion value>,
    promotion_percent: <promotion_percent>,
    status: True,
    expiry: <expiration date>
  },
  . 
  .
  .
  {
    id: <promotion id>,
    name: <promotion name>,
    type: <promotion type>,
    description: <promotion description>,
    promotion_value: <promotion value>,
    promotion_percent: <promotion_percent>,
    status: True,
    expiry: <expiration date>
  }] 
}
```

### PR DESCRIPTION
**Pull request by Awesalot**
- Ability to fetch promotions using status
- 95% coverage
- 10.0 Pylint score

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -194,3 +194,13 @@ def find_by_name(cls, name):
         """
         logger.info("Processing name query for %s ...", name)
         return cls.query.filter(cls.name == name)
+
+    @classmethod
+    def find_by_status(cls, status):
```
Comment (Aman-Chopra) [inline service/models.py]: def find_by_status(cls, status) -> list:

File: service/models.py [hunk 2]
```diff
@@ -194,3 +194,13 @@ def find_by_name(cls, name):
...
+        """Returns all Promotions with the given status
```
Comment (Aman-Chopra) [inline service/models.py]: """Returns all Promotions by their status
:param status: True for promotions that are active
:type available: bool
:return: a collection of Promotions that are active or inactive based on given status
:rtype: list
"""

File: service/routes.py [hunk 3]
```diff
@@ -57,7 +57,12 @@ def index():
 def list_promotions():
     """Returns a list of all of the Promotions"""
     app.logger.info("Request for promotion list")
-    promotions = Promotion.all()
+
```
Comment (Aman-Chopra) [inline service/routes.py:60]: promotions = []

**Review by Aman-Chopra — CHANGES_REQUESTED** [review]
Suggested some additions, please add them and raise another pull request

File: service/routes.py (hunk 3 above)
Comment (Aman-Chopra) [inline service/routes.py:60]: Add promotions = [] before status_type

**Comment by Awesalot** [comment]
Deleted find_by_name as the functionality is currently not required

**Review by Aman-Chopra — APPROVED** [review]
LGTM!

**Review by Black-Caesaris — APPROVED** [review]
LGTM!

**Pull request merged**